docs/old-reference-guide/modules/axon-framework-commands/pages/modeling/command-model.adoc

```diff
@@ -0,0 +1,236 @@
+
+[source,java]
+----
+public record GiftCard(String id, int remainingValue) {
+    @EntityCreator
+    public GiftCard(CardIssuedEvent event) {
+        this(event.cardId(), event.amount());
+    }
+
+    @CommandHandler
+    public static String handle(IssueCardCommand cmd, EventAppender appender) {
+        appender.append(new CardIssuedEvent(cmd.cardId(), cmd.amount()));
+        return cmd.cardId();
+    }
+
+    @CommandHandler
+    public GiftCard handle(RedeemCardCommand cmd, EventAppender appender) {
+        if (cmd.amount() > remainingValue) {
+            throw new IllegalStateException("Insufficient balance");
+        }
+        appender.append(new CardRedeemedEvent(id, cmd.transactionId(), cmd.amount()));
+        return this; // Return current state
+    }
+
+    @EventSourcingHandler
+    public GiftCard on(CardRedeemedEvent event) {
+        return new GiftCard(id, remainingValue - event.amount());
+    }
+}
+----
+
+With immutable entities, `@EventSourcingHandler` methods return a new instance rather than modifying state.
+====
+
+== Entity lifecycle operations
+
+There are several operations that can be performed during the lifecycle of an Entity.
+
+=== Publishing events
+
+Events are published using the `EventAppender` parameter that is injected into command handlers:
+
+[source,java]
+----
+@CommandHandler
+public void handle(UpdateCardCommand cmd, EventAppender appender) {
+    // Validate business rules
+    if (cmd.newAmount() > maximumAllowed) {
+        throw new IllegalArgumentException("Amount exceeds maximum");
+    }
+
+    // Publish event
+    appender.append(new CardUpdatedEvent(id, cmd.newAmount()));
+}
+----
+
+The `EventAppender` publishes events within the scope of the entity executing the operation.
+Events can be published with additional metadata using `appender.append(event, metadata)`.
+
+=== Entity state lifecycle
+
+Entity lifecycle management in Axon 5 is handled through the processing context and entity state:
+
+. **Detecting replay vs live events**: Check whether the Entity is processing live events or being reconstructed from history.
+Use the `ProcessingContext` parameter to determine if an event is being replayed.
+This allows you to perform operations that should only occur when handling newly generated events (for example, sending external notifications).
+
+. **Marking entity as deleted**: Entity deletion is managed through the entity's lifecycle.
+When an entity should no longer accept commands, publish a deletion event and the framework will handle the entity's lifecycle accordingly.
+
+[source,java]
+----
+@CommandHandler
+public void handle(CloseAccountCommand cmd, EventAppender appender) {
+    appender.append(new AccountClosedEvent(id));
+}
+
+@EventSourcingHandler
+public void on(AccountClosedEvent event) {
+    // Entity is marked as deleted by the framework when appropriate
+    // No explicit markDeleted() call needed in Axon 5
+}
+----
```

Comment:
How is marking an entity as deleted or marking an event being a deletion event done? I think there is there something missing in this description.